### ISSUE TITLE
learn.predict() runs into an Exception Value error

### ISSUE THREAD

**Issue by harshikanu** [fastai/fastai#2580]
am trying to predict the sentiment of a sentence by using learn.predict(message). I am loading a pretrained Databunch and model files. 
But I am getting the following error -

Code - 
   `final_result = []`
   `data = load_data(_path,
                     '/home/data_save.pkl')`
    `roberta_model = CustomRobertaModel()`
    `learn = Learner(data, roberta_model, metrics=[accuracy])`
    `st2 = torch.load("/home/final_model_base.pth",
                     map_location=torch.device('cpu')
                     )`
    `learn.model.load_state_dict(st2)`
    `learn.model.roberta.eval()`
    

ERROR -

File “/usr/local/lib/python3.7/site-packages/fastai/basic_train.py” in predict
382. y = ds.y.reconstruct(pred, x) if has_arg(ds.y.reconstruct, ‘x’) else ds.y.reconstruct(pred)

File “/usr/local/lib/python3.7/site-packages/fastai/text/data.py” in reconstruct
341. return Text(t[idx_min:idx_max+1], self.vocab.textify(t[idx_min:idx_max+1]))

File “/usr/local/lib/python3.7/site-packages/fastai/text/transform.py” in textify
134. return sep.join([self.itos[i] for i in nums]) if sep is not None else [self.itos[i] for i in nums]

File “/usr/local/lib/python3.7/site-packages/fastai/text/transform.py” in
134. return sep.join([self.itos[i] for i in nums]) if sep is not None else [self.itos[i] for i in nums]

Exception Type: TypeError at /test_sent/
Exception Value: only integer tensors of a single element can be converted to an index

Running pred_batch or get_preds works. But I want to run it with .predict since it is more efficient. 

Please help me with this.

**Comment by sgugger** [comment]
Please use the [forum](https://forums.fast.ai/) to help debug your code. We keep the issues for known bugs only.
